pkg/kubelet/cm/helpers_linux.go

```diff
@@ -44,3 +47,46 @@ func GetCgroupSubsystems() (*CgroupSubsystems, error) {
 		MountPoints: mountPoints,
 	}, nil
 }
+
+// GetPodResourceRequest returns the pod requests for the supported resources.
+// Pod request is the summation of resource requests of all containers in the pod.
+func GetPodResourceRequests(pod *api.Pod) api.ResourceList {
+	requests := api.ResourceList{}
+	for _, container := range pod.Spec.Containers {
+		requests = quota.Add(requests, container.Resources.Requests)
+	}
+	return requests
+}
+
+// GetPodResourceLimits returns the pod limits for the supported resources
+// Pod limit is the summation of resource limits of all containers
+// in the pod. If limit for a particular resource is not specified for
+// even a single container then we return the node resource capacity
+// as the pod limit for the particular resource.
+func GetPodResourceLimits(pod *api.Pod, nodeInfo *api.Node) api.ResourceList {
+	capacity := nodeInfo.Status.Capacity
+	limits := api.ResourceList{}
+	zeroQuantity := resource.MustParse("0")
+	for _, resource := range []api.ResourceName{api.ResourceCPU, api.ResourceMemory} {
+		for _, container := range pod.Spec.Containers {
+			quantity := container.Resources.Limits[resource]
```

Comment:
You could also check if the quantity exists first before checking if its zero. I'm not sure if our defaulting logic adds a zero quantity whenever limits are not specified. In any case, it is better to stick to golang semantics for maps. Check that a value exists before using it.